cmd/internal/cli/inspect_linux.go

```diff
@@ -257,40 +271,130 @@ var InspectCmd = &cobra.Command{
 	Example: docs.InspectExample,
 
 	Run: func(cmd *cobra.Command, args []string) {
-		// Sanity check
-		if _, err := os.Stat(args[0]); err != nil {
-			sylog.Fatalf("container not found: %s", err)
+		sandboxImage := false
+		a := []string{"/bin/sh", "-c", ""}
+
+		f, err := os.Stat(args[0])
+		if os.IsNotExist(err) {
+			sylog.Fatalf("Container not found: %s\n", err)
+		} else if err != nil {
+			sylog.Fatalf("Unable to stat file: %s", err)
+		}
+		if f.IsDir() {
+			sandboxImage = true
+		}
+
+		var fimg sif.FileImage
+		if !sandboxImage {
+			var err error
+			fimg, err = sif.LoadContainer(args[0], true)
+			if err != nil {
+				sylog.Fatalf("failed to load SIF container file: %s", err)
+			}
+			defer fimg.UnloadContainer()
+		}
+
+		var inspectData inspectFormat
+		inspectData.Type = "container"
+		inspectData.Data.Attributes.Labels = make(map[string]string, 1)
+
+		inspectLabelInContainer := func() {
+			sylog.Debugf("Inspection of labels selected.")
+			a[2] += getLabelsCommand()
+		}
+		// Inspect Labels.
+		if labels || defaultToLabels() {
+			jsonName := ""
+			if AppName == "" {
+				jsonName = "system-partition"
+			} else {
+				jsonName = AppName
+			}
+
+			if sandboxImage {
+				sylog.Debugf("Inspecting in the container...")
```

Comment:
In sandbox maybe?

Reply:
Not just for sandbox, but if your inspecting a image without a label partition.